### PR TITLE
use Autotools on non-Windows builds.

### LINKED ISSUE #13: 0.2 package does not create `libyaml-0.so.2` so link
Issue:

The 0.2.2 version of **yaml** package does not create a symbolic link for `libyaml-0.so.2` (as the version 0.1.7 does). So some of dependents are not working with the latest version of **yaml**.

From what I checked, latest **uwsgi** package and some old version  of **pyyaml** are throwing the following error: `error while loading shared libraries: libyaml-0.so.2: cannot open shared object file: No such file or directory`

By checking the yaml package from other providers, it seems that `libyaml-0.so.2` should be there:
- https://www.archlinux.org/packages/extra/x86_64/libyaml/files/
- https://pkgs.alpinelinux.org/contents?branch=edge&name=yaml&arch=x86_64&repo=main

<br/>
Environment (<code>conda list</code>):
<details>

```
$ conda list
# Name                    Version                   Build  Channel
_libgcc_mutex             0.1                        main    conda-forge
ca-certificates           2019.11.28           hecc5488_0    conda-forge
certifi                   2019.11.28               py36_0    conda-forge
icu                       64.2                 he1b5a44_1    conda-forge
jansson                   2.11              h516909a_1001    conda-forge
libffi                    3.2.1             he1b5a44_1006    conda-forge
libgcc-ng                 9.2.0                hdf63c60_0    conda-forge
libiconv                  1.15              h516909a_1005    conda-forge
libstdcxx-ng              9.2.0                hdf63c60_0    conda-forge
libxml2                   2.9.10               hee79883_0    conda-forge
ncurses                   6.1               hf484d3e_1002    conda-forge
openssl                   1.0.2t               h14c3975_0    conda-forge
pcre                      8.43                 he1b5a44_0    conda-forge
pip                       19.3.1                   py36_0    conda-forge
python                    3.6.7             hd21baee_1002    conda-forge
readline                  7.0               hf8c457e_1001    conda-forge
setuptools                42.0.2                   py36_0    conda-forge
sqlite                    3.28.0               h8b20d00_0    conda-forge
tk                        8.6.10               hed695b0_0    conda-forge
uwsgi                     2.0.17.1         py36h2184488_1    conda-forge
wheel                     0.33.6                   py36_0    conda-forge
xz                        5.2.4             h14c3975_1001    conda-forge
yaml                      0.2.2                he1b5a44_0    conda-forge
zlib                      1.2.11            h516909a_1006    conda-forge
```
</details>

<br/>
Details about  <code>conda</code> and system ( <code>conda info</code> ):
<details>

```
$ conda info

     active environment : yaml-test
    active env location : /opt/conda/envs/yaml-test
            shell level : 2
       user config file : /root/.condarc
 populated config files : /root/.condarc
          conda version : 4.6.14
    conda-build version : not installed
         python version : 3.7.3.final.0
       base environment : /opt/conda  (writable)
           channel URLs : https://conda.anaconda.org/conda-forge/linux-64
                          https://conda.anaconda.org/conda-forge/noarch
                          https://repo.anaconda.com/pkgs/main/linux-64
                          https://repo.anaconda.com/pkgs/main/noarch
                          https://repo.anaconda.com/pkgs/free/linux-64
                          https://repo.anaconda.com/pkgs/free/noarch
                          https://repo.anaconda.com/pkgs/r/linux-64
                          https://repo.anaconda.com/pkgs/r/noarch
          package cache : /opt/conda/pkgs
                          /root/.conda/pkgs
       envs directories : /opt/conda/envs
                          /root/.conda/envs
               platform : linux-64
             user-agent : conda/4.6.14 requests/2.21.0 CPython/3.7.3 Linux/4.15.0-70-generic debian/9 glibc/2.24
                UID:GID : 0:0
             netrc file : None
           offline mode : False


```
</details>

### PR DESCRIPTION
**Pull request by seanyen**
Checklist
* [x] Used a [fork of the feedstock to propose changes](https://conda-forge.org/docs/maintainer/updating_pkgs.html#forking-and-pull-requests)
* [x] Bumped the build number (if the version is unchanged)
* [x] Reset the build number to `0` (if the version changed)
* [x] [Re-rendered]( https://conda-forge.org/docs/maintainer/updating_pkgs.html#rerendering-feedstocks ) with the latest `conda-smithy` (Use the phrase <code>@<space/>conda-forge-admin, please rerender</code> in a comment in this PR for automated rerendering)
* [x] Ensured the license file is being packaged.

This fixes #13

### CODE REVIEW

File: recipe/meta.yaml [hunk 1]
```diff
@@ -21,6 +21,10 @@ requirements:
     - {{ compiler('c') }}                # [not win]
     - vs2017_win-64                      # [win64]
     - vs2017_win-32                      # [win32]
+    - automake                           # [not win]
+    - autoconf                           # [not win]
+    - make                               # [not win]
+    - libtool                            # [not win]
     - cmake
```
Comment (mbargull) [inline recipe/meta.yaml]: You can add the `[win]` selector to `cmake`, `pkg-config`, `ninja`.

File: recipe/meta.yaml [hunk 2]
```diff
@@ -21,6 +21,10 @@ requirements:
     - {{ compiler('c') }}                # [not win]
     - vs2017_win-64                      # [win64]
     - vs2017_win-32                      # [win32]
```
Comment (mbargull) [inline recipe/meta.yaml]:
```suggestion
```
We generally don't support `win32` anymore.

File: recipe/meta.yaml [hunk 3]
```diff
@@ -21,6 +21,10 @@ requirements:
     - {{ compiler('c') }}                # [not win]
     - vs2017_win-64                      # [win64]
```
Comment (mbargull) [inline recipe/meta.yaml]: Does this really need 2017 or is 2015 fine, too? If 2015 works, we can use `{{ compiler('c') }}` on all platforms.

Comment (mbargull) [reply recipe/meta.yaml]: Also, this is C-only, right? Then please remove `{{ compiler('cxx') }}`.

**Review by mbargull — CHANGES_REQUESTED** [review]
Only small remarks, but generally, this works on a local build:
```sh
ls -lA lib/*.so*
lrwxrwxrwx 1 conda conda     18 Dec  2 20:26 lib/libyaml-0.so.2 -> libyaml-0.so.2.0.6
-rwxrwxr-x 1 conda conda 125064 Dec  2 20:26 lib/libyaml-0.so.2.0.6
lrwxrwxrwx 1 conda conda     18 Dec  2 20:26 lib/libyaml.so -> libyaml-0.so.2.0.6
```

Thanks!

**Comment by seanyen** [comment]
@mbargull Thanks for the comments. I fixed them and let's wait for the CI result.

**Review by mbargull — COMMENTED** [review]
Looks good, thanks @seanyen!

**Pull request merged**